### PR TITLE
Sk tc filter shopping list

### LINKED ISSUE #9: 9. As a user, I want to filter my shopping list to make it easier to locate an item in the list.
AC:

* Display a text field above the top of the shopping list
* As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
* When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
* The filter text should match any part of the item name (i.e. it should _not_ only match from the start of the string)

Notes:

* The wireframe shows the filter field, but not it being used. https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png
* The following example app from MDN is a great example of providing feedback that is accessible to screenreaders when things change on the screen. https://moz-todomvc-react.netlify.app/

### PR DESCRIPTION
**Pull request by AirZT11**
## Description

* Created an input field in the GroceryContainer component to filter the list based on user input
* Displayed a input field reset button when input field was populated
* Displayed a loading prompt when the grocery state was empty and the async call to database was still loading
* Displayed bottomNav or create-list/submit-token based on loggedIn status

## Related Issue

closes #9 

## Acceptance Criteria

[AC](https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/9)

As a user, I want to filter my shopping list to make it easier to locate an item in the list

* Display a text field above the top of the shopping list
* As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
* When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
* The filter text should match any part of the item name (i.e. it should not only match from the start of the string)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with git pull origin sk-tc-filter-shopping-list and check that branch out with git checkout sk-tc-filter-shopping-list

* First log in or create list, then make sure that there is an input field for 'Search Field'
* Type in the input field and make sure that the list is filtered based on input
* Once input is populated, make sure an 'X' button appears and clears the field when clicked
* When logged out, user should only see the submit token and create list fields.
* When logged in, user should see the bottomNav

### CODE REVIEW

File: src/components/containers/GroceryContainer.js [hunk 1]
```diff
@@ -1,10 +1,14 @@
 import React, { useState, useEffect } from 'react';
 import { fb } from '../../lib/firebase';
 import GroceryCard from '../GroceryCard';
+import BottomNav from '../BottomNav';
 import { useHistory } from 'react-router-dom';
 
 const GroceryContainer = () => {
   const [grocery, setGrocery] = useState([]);
+  const [loading, setLoading] = useState(true);
```
Comment (Dani-Kirby) [inline src/components/containers/GroceryContainer.js:9]: Ooooh loading state. I know we talked about this week one, so happy to see it coming into play now.

File: src/components/containers/GroceryContainer.js [hunk 2]
```diff
@@ -30,34 +34,66 @@ const GroceryContainer = () => {
...
-          {grocery.map((g) => (
-            <li key={g.itemName}>
-              <GroceryCard item={g} />
-            </li>
-          ))}
+          {grocery
+            .filter((g) =>
+              g.itemName.toLowerCase().includes(input.toLowerCase()),
+            )
+            .map((g) => (
```
Comment (Dani-Kirby) [inline src/components/containers/GroceryContainer.js:90]: ooh .filter is a great method for this. 
Do we need to think about spaces as well? for instance something like 'cream cheese' should we populate results if the user types 'creamcheese' or is this overthinking

File: src/components/containers/GroceryContainer.js [hunk 3]
```diff
@@ -30,34 +34,66 @@ const GroceryContainer = () => {
...
-  if (grocery.length === 0) {
+  const handleChange = (e) => {
+    setInput(e.target.value);
+    e.target.value ? setResetDisplay('inline') : setResetDisplay('none');
+  };
+
+  const handleResetClick = () => {
+    setInput('');
+    setResetDisplay('none');
+  };
```
Comment (Dani-Kirby) [inline src/components/containers/GroceryContainer.js:53]: I love simple and clean functions like this. No frills, just gets the specific job done.

**Review by Dani-Kirby — APPROVED** [review]
Filtering is working well during my testing. Love this functionality. We are seriously getting close to a solid app here!
We might want to consider more functionality in the search field around spaces or punctuation, but we could probably make an argument for either direction.

**Review by YolandaHaynes — COMMENTED** [review]
The filter feature is cool! You guys did an awesome job, and it worked on my end! I really like that it populated the results while I was typing. I think it was a great idea to move the bottom navigation component.

**Comment by lyleschemmerling** [comment]
It's really nice when you type in a filter box and it is snappy and responsive. I love it.

File: src/components/containers/GroceryContainer.js (hunk 2 above)
Comment (lyleschemmerling) [reply src/components/containers/GroceryContainer.js:90]: I like this thought, although as a user I don't think that I would *expect* it to handle spaces so I think its fine as is.

**Review by lyleschemmerling — APPROVED** [review]
This looks great

**Pull request merged**